sql/analyzer/analyzer.go

```diff
@@ -395,6 +462,35 @@ func (a *Analyzer) analyzeWithSelector(ctx *sql.Context, n sql.Node, scope *Scop
 	return n, err
 }
 
+func (a *Analyzer) analyzeWithExceptor(ctx *sql.Context, n sql.Node, scope *Scope, except map[string]struct{}) (sql.Node, error) {
+	span, ctx := ctx.Span("analyze", opentracing.Tags{
+		//"plan": , n.String(),
+	})
+
+	var err error
+	a.Log("starting analysis of node of type: %T", n)
+	for _, batch := range a.Batches {
+		batch.Skip = except
```

Comment:
Rather than this, write EvalWithSelector and pass a function based on the skip set